backend/dgraph/src/entities.rs

```diff
@@ -41,17 +80,117 @@ query EntitiesQuery($search: String = "", $first: Int = 10, $offset: Int = 0) {
 #[cfg(test)]
 mod tests {
 
+    use std::vec;
+
     use super::*;
 
     #[tokio::test]
     async fn test_search() {
         let client = DgraphClient::new("http://localhost:8080/graphql");
         let variables = SearchVars {
-            search: "/Heparin.Sodium.*/".to_string(),
-            first: 10,
-            offset: 0,
+            filter: DgraphFilter {
+                description: Some(DgraphFilterType {
+                    regexp: Some("/Heparin Sodium.*/".to_string()),
+                    ..Default::default()
+                }),
+                category: Some(DgraphFilterType {
+                    r#in: Some(vec!["Drug".to_string()]),
+                    ..Default::default()
```

Comment:
Some helper functions like we have for our filters e.g. 
`let permission_filter = UserPermissionFilter::new().user_id(EqualFilter::equal_to(&ctx.user_id));`

I think we can wait to see if that's helpful later on? Or I could add to this PR?

Reply:
Ah yeah, those filters are nice, but can be added later for sure :)